feature(Job Title Management)

lomamech opened the pull request:
- Implement Job Title(Function) Client Side,
- Implement test E2E for Job title management module


reference #2292

jniles commented on client/src/i18n/en/form.json:
```diff
@@ -312,7 +312,7 @@
             "FROM": "From",
             "FOR": "For",
-            "FUNCTION": "Function",
+            "FUNCTION": "Job Title",
```
This is a much nicer name!  :+1:

lomamech committed:
Rebase with branch Master

jniles commented on client/src/i18n/fr/function.json:
```diff
@@ -0,0 +1,18 @@
+{
+  "FUNCTION" : {
```
Since this is no longer translated as "Function", we should make all the keys "PROFESSION" or something similar.  This can be a separate PR if you wish.

jniles commented on client/src/modules/functions/functions.js:
```diff
@@ -0,0 +1,98 @@
+      vm.gridOptions.data = data;
+    })
+    .catch(Notify.handleError)
+    .finally(function () {
```
Nice :+1:

jniles commented on client/src/modules/functions/functions.js:
```diff
@@ -0,0 +1,98 @@
+  }
+
+  // switch to delete warning mode
+  function deleteFunction(title) {
```
Instead of "title" which sounds like it might be a `String`, try calling this `profession`.  I thought it was a text string at first.

... alternatively, just pass in the `id`.  It looks like all you actually need is the `id`.

jniles commented on test/end-to-end/functions/functions.spec.js:
```diff
@@ -0,0 +1,39 @@
+/** configuring helpers**/
+helpers.configure(chai);
+
+describe('Functions Management', () => {
```
This should be called `Job Titles Management` now, right?

jniles reviewed (approved):
@lomamech, this is pretty good code!

I've made a few comments - none of them are blocking.  Would you like to work on them before a merge or would you like me to merge these and make an issue tracking changes?

Great work!

lomamech committed:
perf(Job Title Management)
- Use PROFESSION instead FUNCTION
- Update describe text in test E2E job titles management text

jniles reviewed (approved):
LGTM.

bors r+
